subworkflows/local/amp.nf

```diff
@@ -0,0 +1,115 @@
+        ch_ampresults_for_ampcombi = ch_ampresults_for_ampcombi.mix(AMPIR.out.amps_tsv)
+    }
+
+    // HMMSEARCH
+    if ( !params.amp_skip_hmmsearch ) {
+        if ( params.amp_hmmsearch_models ) { ch_amp_hmm_models = Channel.fromPath( params.amp_hmmsearch_models, checkIfExists: true ) } else { exit 1, '[nf-core/funcscan] error: hmm model files not found for --amp_hmmsearch_models! Please check input.' }
+
+        ch_amp_hmm_models_meta = ch_amp_hmm_models
+            .map {
+                file ->
+                    def meta  = [:]
+                    meta['id'] = file.extension == 'gz' ? file.name - '.hmm.gz' :  file.name - '.hmm'
+
+                [ meta, file ]
+            }
+
+        ch_in_for_amp_hmmsearch = ch_faa_for_amp_hmmsearch.combine(ch_amp_hmm_models_meta)
+            .map {
+                meta_faa, faa, meta_hmm, hmm ->
+                    def meta_new = [:]
+                    meta_new['id']     = meta_faa['id']
+                    meta_new['hmm_id'] = meta_hmm['id']
+                [ meta_new, hmm, faa, params.amp_hmmsearch_savealignments, params.amp_hmmsearch_savetargets, params.amp_hmmsearch_savedomains ]
+            }
+
+        AMP_HMMER_HMMSEARCH ( ch_in_for_amp_hmmsearch )
+        ch_versions = ch_versions.mix(AMP_HMMER_HMMSEARCH.out.versions)
+// This chunk of code can add the hmmer_hmmsearch channel in the ampcombi input channel `ch_ampcombi_input`.
+// Currently deactivated as AMPcombi can only accept a single HMM file (whereas funcscan produces multiple, one per each input HMM).
+//        GUNZIP_HMMER ( AMP_HMMER_HMMSEARCH.out.output )
+//        ch_hmmout = GUNZIP_HMMER.out.gunzip
+//                        .map {
+//                            meta_id, meta_hmm ->
+//                            def meta_hmmsearch = [:]
+//                            meta_hmmsearch['id'] = meta_id['id']
+//                            [meta_hmmsearch, meta_hmm]
+//                        }
+//
+//        ch_ampresults_for_ampcombi = ch_ampresults_for_ampcombi.mix(ch_hmmout)
+    }
+
+    //AMPCOMBI
+    ch_input_for_ampcombi = ch_ampresults_for_ampcombi
+        .groupTuple()
+        .join( ch_faa_for_ampcombi )
+        .multiMap{
+            input: [ it[0], it[1] ]
+            faa: it[2]
+        }
+    // Checks if `--amp_database` is a user supplied path and if the path does not exist it goes to default, which downloads the DRAMP database once.
+    if ( params.amp_ampcombi_db ) { ch_ampcombi_input_db = Channel.fromPath( params.amp_ampcombi_db, checkIfExists: true ) } else { ch_ampcombi_input_db = [] }
+
+    AMPCOMBI( ch_input_for_ampcombi.input, ch_input_for_ampcombi.faa, ch_ampcombi_input_db )
+    ch_ampcombi_summaries = ch_ampcombi_summaries.mix(AMPCOMBI.out.csv)
+
+    //AMPCOMBI concatenation
+    ch_ampcombi_summaries_out = ch_ampcombi_summaries
+        .multiMap{
+                input: [ it[0] ]
+                summary: it[1]
+            }
```

Comment:
Why a multiap rather than just a map that strips out the `input` and keeps `summary?

Reply:
Actual nm, possibly ineffecient but it works ;)